packages/terra-functional-testing/src/commands/viewport-helpers.js

```diff
@@ -0,0 +1,71 @@
+const Logger = require('../logger/logger');
+const terraViewports = require('../util/viewports');
+
+const logger = new Logger({ prefix: 'wdio-terra-service' });
+
+/**
+* Convenience method for getting the Terra test viewports by name.
+*
+* @param {string|string[]} sizes - the list of Terra viewport names.
+* @return {object} - Terra viewport sizes.
+*/
+const getViewports = (...sizes) => {
+  let viewportSizes = Object.keys(terraViewports);
+  if (sizes.length) {
+    viewportSizes = sizes;
+  }
+  return viewportSizes.map(formFactor => terraViewports[formFactor]);
+};
+
+/**
+* Sets the viewport for the test run to the Terra test viewport size.
+*
+* @param {string} [formFactor=huge] - the Terra test viewport.
+*/
+const setViewport = (formFactor = 'huge') => {
+  const terraViewport = terraViewports[formFactor];
+  if (terraViewport !== undefined && typeof terraViewport === 'object') {
+    global.browser.setWindowSize(terraViewport.width, terraViewport.height);
+  } else {
+    throw logger.error(`The ${formFactor} formFactor supplied is not a viewport size supported by Terra.`);
+  }
+};
+
+/**
+* Mocha `describe` block to set and loop the tests viewports to simplify writing tests.
+* If defined, the formFactor will be used if it is included in the list of test viewports.
+* Otherwise, use the list of test viewports.
+*
+* This is intended to be used as a root-level Mocha `describe`.
+*
+* @param {string} title - The `describe` block title.
+* @param {string[]} viewports - The list of Terra viewports to tests if formFactor is not set.
+* @param {function} - the test function to execute for each viewport.
+*/
+const describeViewports = (title, viewports, fn) => {
+  let localViewports = viewports;
+
+  // If formFactor is defined and viewports contains the form factor, run that size else don't run the tests
+  const { formFactor } = global.browser.config;
+  if (formFactor) {
+    localViewports = viewports.includes(formFactor) ? [formFactor] : [];
+  }
+
+  let currentViewportSize;
+  localViewports.forEach(viewport => global.describe(`[${viewport}]`, () => {
```

Comment:
Doesn't matter, but we should consider using [describe.each](https://jestjs.io/docs/en/api#describeeachtablename-fn-timeout)